### PR TITLE
Issue 7

### PR DESCRIPTION
**Pull request by Pasqua101**
Completed implementation of .md conversion.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Learn2Blog
 
-This is an open source command line app to help you quickly convert your text notes into html files so they can be easily published on your blog.
-With this tool, you can write your notes in plain text files and convert them into html files with a single command. This way, you can focus on learning to blog and not worry about the technical details of creating a blog.
+This is an open source command line app to help you quickly convert your text and md notes into html files so they can be easily published on your blog.
```
Comment (Yousef-Majidi) [inline README.md]: md -> markdown

File: README.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
+With this tool, you can write your notes in plain text or md files and convert them into html files with a single command. This way, you can focus on learning to blog and not worry about the technical details of creating a blog.
```
Comment (Yousef-Majidi) [inline README.md]: md -> markdown

File: README.md [hunk 3]
```diff
@@ -73,10 +73,17 @@ Run the app by using one of the options or provide an input argument:
 ```
 
 ```bash
-# converts all .txt files in the directory into html and outputs in the default directory, unless specified with the -o flag
+# converts all .txt and md files in the directory into html and outputs in the default directory, unless specified with the -o flag
```
Comment (Yousef-Majidi) [inline README.md]: md -> .md

File: src/MdProcessor.cs
Comment (Yousef-Majidi) [inline src/MdProcessor.cs:1]: This file contains a lot of copied code. In the spirit of not writing repeated code, consider how you could change this so that you're utilizing the existing code as much as possible.
I would suggest removing the class `MdProcessor` entirely, and simply adding the function `ConvertMdToHtml` as a member of the class `HtmlProcessor`. 

function `ProcessFile` is entirely unnecessary to exist as a copy in two classes. Everything is a public member, so there is no access problem for any part of the code that needs to use them.

File: src/MdProcessor.cs [hunk 4]
```diff
@@ -0,0 +1,206 @@
+﻿using System;
+using System.IO;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Learn2Blog.src
+{
+    public partial class MdProcessor
+    {
+        /* public static string ConvertMdToHtml(string title, string text)
```
Comment (Yousef-Majidi) [inline src/MdProcessor.cs]: remove this comment block before the final merge. but I like suggested, remove this class and .cs file altogether and add the new function below to `HtmlProcessor` class.

File: src/MdProcessor.cs [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+
+             htmlBuilder.AppendLine("</html>");
+             // HTML end
+
+             return htmlBuilder.ToString();
+         }*/
+
+        public static string ConvertMdToHtml(string title, string text)
+        {
+            int titleStart = text.IndexOf("\r\n\r\n\r\n");
```
Comment (Yousef-Majidi) [inline src/MdProcessor.cs]: this section of the code is only for detecting a title in the text file. Markdown as a specific syntax for a title. Please remove anything related to detecting or adding the title to the HTML from this function

File: src/Program.cs [hunk 6]
```diff
@@ -1,4 +1,6 @@
-﻿namespace Learn2Blog
+﻿using Learn2Blog.src;
```
Comment (Yousef-Majidi) [inline src/Program.cs]: No need for this

File: src/Program.cs [hunk 7]
```diff
@@ -26,21 +28,51 @@ static void Main(string[] args)
                 if (File.Exists(inputPath))
                 {
                     CommandLineUtils.CreateOutputDirectory(outputPath);
-                    HtmlProcessor.ProcessFile(inputPath, outputPath);
+
+                    // Check file extension to decide whether to use MdProcessor or HtmlProcessor
+                    string fileExtension = Path.GetExtension(inputPath);
+                    if (fileExtension == ".md")
+                    {
+                        MdProcessor.ProcessFile(inputPath, outputPath);
```
Comment (Yousef-Majidi) [inline src/Program.cs]: The class `HtmlProcessor` should be only class handling this. But there should be two separate functions for handling `.md` versus `.txt`. The `ProcessFile` function should be called from `HtmlProcessor`. This function however should be modified to contain logic for handling `.txt` versus `.md`. `ProcessFile` is the decider of which processor function to call.

File: src/Program.cs [hunk 8]
```diff
@@ -26,21 +28,51 @@ static void Main(string[] args)
...
+                    }
+                    else if (fileExtension == ".txt")
+                    {
+                        HtmlProcessor.ProcessFile(inputPath, outputPath);
+                    }
+                    else
+                    {
+                        CommandLineUtils.Logger($"Unsupported file format for input path: {inputPath}");
```
Comment (Yousef-Majidi) [inline src/Program.cs]: This is check is unnecessary, because it already exists in `HtmlProcessor`. But you should modify it to check for `.md` as well.

File: src/Program.cs [hunk 9]
```diff
@@ -26,21 +28,51 @@ static void Main(string[] args)
...
+                    }
                 }
                 else if (Directory.Exists(inputPath))
                 {
-                    string[] files = Directory.GetFiles(inputPath, "*.txt");
+                    // Get all files in the directory and save them to the files array
+                    string[] files = Directory.GetFiles(inputPath, "*.*");
```
Comment (Yousef-Majidi) [inline src/Program.cs]: Should not be allowing "\*.\*". The program should _only_ accept ".txt" or ".md". please revert this change and add an `or` condition for ".md"

File: src/Program.cs [hunk 10]
```diff
@@ -26,21 +28,51 @@ static void Main(string[] args)
...
                 else if (Directory.Exists(inputPath))
                 {
-                    string[] files = Directory.GetFiles(inputPath, "*.txt");
+                    // Get all files in the directory and save them to the files array
+                    string[] files = Directory.GetFiles(inputPath, "*.*"); 
+
                     if (files.Length == 0)
                     {
-                        CommandLineUtils.Logger($"No txt files found in directory {inputPath}");
+                        CommandLineUtils.Logger($"No files found in directory {inputPath}");
```
Comment (Yousef-Majidi) [inline src/Program.cs]: revert this change. But add ".txt or .md" files..."

File: src/Program.cs [hunk 11]
```diff
@@ -26,21 +28,51 @@ static void Main(string[] args)
...
+                        {
+                            MdProcessor.ProcessFile(file, outputPath);
+                        }
+                        else if (fileExtension == ".txt")
+                        {
+                            HtmlProcessor.ProcessFile(file, outputPath);
+                        }
+                        else
+                        {
+                            CommandLineUtils.Logger($"Unsupported file format for file: {file}");
```
Comment (Yousef-Majidi) [inline src/Program.cs]: Unnecessary because before we reach this point of the code, we already only retrieve .md or .txt files

**Review by Yousef-Majidi — CHANGES_REQUESTED** [review]

File: src/HtmlProcessor.cs [hunk 12]
```diff
@@ -58,10 +58,87 @@ public static string ConvertTextToHtml(string title, string text)
...
-            if (Path.GetExtension(inputPath) != ".txt")
+            
+            if (Path.GetExtension(inputPath) == ".txt" || Path.GetExtension(inputPath) == ".md")
+            {
+                //Do nothing
+            }
+            // if the file is not a text or markdown file, log and return
+            else
             {
                 CommandLineUtils.Logger($"The specified input file [{inputPath}] is not a text file");
```
Comment (Yousef-Majidi) [inline src/HtmlProcessor.cs]: The error message should update to "... is not a text or markdown file"

File: src/HtmlProcessor.cs [hunk 13]
```diff
@@ -70,11 +147,20 @@ public static void ProcessFile(string inputPath, string outputPath)
...
+
+                if (Path.GetExtension(inputPath) == ".txt")
+                {
+                    html = ConvertTextToHtml(Path.GetFileNameWithoutExtension(inputPath), text);
+
+                }
+                if (Path.GetExtension(inputPath) == ".md")
+                {
+                    html = ConvertMdToHtml(Path.GetFileNameWithoutExtension(inputPath), text);
+                }
```
Comment (Yousef-Majidi) [inline src/HtmlProcessor.cs]: store the extension in a variable before this block, and use the variable for the following if checks. That way, we don't have to call the GetExtension function twice.
Alternatively, change the block to an if-else statement, because a file can only be .txt or .md. If it's not one, it's the other. Since by the time we have reached this point in the code, we have already checked for those two. No other file type would make the code reach this block

File: src/HtmlProcessor.cs [hunk 14]
```diff
@@ -58,10 +58,87 @@ public static string ConvertTextToHtml(string title, string text)
...
         {
-            // if the file is not a text file, log and return
-            if (Path.GetExtension(inputPath) != ".txt")
+            
+            if (Path.GetExtension(inputPath) == ".txt" || Path.GetExtension(inputPath) == ".md")
+            {
+                //Do nothing
+            }
+            // if the file is not a text or markdown file, log and return
+            else
```
Comment (Yousef-Majidi) [inline src/HtmlProcessor.cs]: "if (condition) { // do nothing }" is not a good programming practice.
You can change this in two ways. Either flip the conditions and log error and return when they fail. Or, put the try-catch below below the else statement inside this if statement.

**Review by Yousef-Majidi — CHANGES_REQUESTED** [review]

File: src/HtmlProcessor.cs (hunk 14 above)
Comment (Pasqua101) [reply src/HtmlProcessor.cs]: I had to put the try catch into the if statement. Otherwise if I tried flipping the conditions, it wouldn't process any files.

File: src/HtmlProcessor.cs (hunk 13 above)
Comment (Pasqua101) [reply src/HtmlProcessor.cs]: Went with the if-else statement instead.

File: src/HtmlProcessor.cs (hunk 12 above)
Comment (Pasqua101) [reply src/HtmlProcessor.cs]: Done.

**Review by Yousef-Majidi — APPROVED** [review]
